include/veriblock/blockchain/block_status.hpp

```diff
@@ -29,12 +25,23 @@ enum BlockStatus : uint32_t {
       BLOCK_FAILED_CHILD | BLOCK_FAILED_POP | BLOCK_FAILED_BLOCK,
   //! the block has been applied via PopStateMachine
   BLOCK_APPLIED = 1 << 5,
-  //! the block has been at least once successful applied via PopStateMachine
-  BLOCK_CAN_BE_APPLIED = 1 << 6,
-  //! if set, acceptBlock has been executed on this block
-  BLOCK_HAS_PAYLOADS = 1 << 7,
+
+  //! acceptBlockHeader succeded. All ancestors are at least at this state.
+  BLOCK_VALID_TREE = 1 << 6,
+  //! acceptBlock has been executed on this block; payloads are statelessly
+  //! valid
+  BLOCK_HAS_PAYLOADS = 2 << 6,
   //! the block is connected via connectBlock
-  BLOCK_CONNECTED = 1 << 8,
+  BLOCK_CONNECTED = 3 << 6,
+  //! the block has been successfully applied, likely along with another chain
+  BLOCK_HAS_BEEN_APPLIED = 4 << 6,
```

Comment:
What's the difference between BLOCK_HAS_BEEN_APPLIED and BLOCK_APPLIED?

Reply:
naming of BLOCK_CAN_BE_APPLIED is also awful. We need to rename them. BLOCK_APPLIED means it is applied right now. BLOCK_HAS_BEEN_APPLIED is a validity status - has been applied before

Reply:
maybe name it `BLOCK_VALID`?